tests/unit/test_beaker.py

```diff
@@ -91,3 +92,72 @@ async def test_beaker_job_creation(self, mock_beaker_conf):
         # so that provisioning won't fail on AVC denials
         xml = job.toxml()
         assert '<param name="RSTRNT_DISABLED" value="10_avc_check"/>' in xml
+
+    @pytest.mark.asyncio
+    async def test_translate_constraint(self, mock_beaker_conf):
+        provider = BeakerProvider()
+        await provider.init(self.distros, self.timeout, self.reserve_duration)
+        req_node = xml_doc().createElement("and")
+        host_requires = {
```

Comment:
In principle, it looks good. But I would change the following:

- Instead of asserting a pretty XML, it should be normalized, and them compared. We could use e.g. minidom. Check an example here https://www.geeksforgeeks.org/compare-two-xml-files-in-python/
- Take host_requires out of the method so it can be reused if needed in future tests. Either as a class variable or a function that returns the object (see other test files)